### PR TITLE
Add prepare-commit-msg hook and guide

### PR DESCRIPTION
**Pull request by rafaelcpalmeida**

### CODE REVIEW

File: hooks/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Automatically add task type to your commit messages
+
+The following guid will show you how to add the task type to your commit messages, automatically, using git hooks.
```
Comment (portellaa) [inline hooks/README.md:3]: guide

File: hooks/prepare-commit-msg [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# List the branch prefixes that apply, bellow
+if [ -z "$BRANCHES_TO_APPLY" ]; then
+	BRANCHES_TO_APPLY=(chore fix feat)
+fi
+
+# Pick the current branch name
+BRANCH_NAME=$(git symbolic-ref --short HEAD)
+
+# Remove the unnecessary parts
+TRIMMED=$(echo $BRANCH_NAME | grep -Eo '^(\w+)' | sed -e 'y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/')
```
Comment (portellaa) [inline hooks/prepare-commit-msg:12]: just a suggestion easier to read and maintain: `$(echo $BRANCH_NAME | cut -d/ -f1)`

File: hooks/prepare-commit-msg [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+# Check if it matches
+APPLY_TO_BRANCH=$(printf "%s\n" "${BRANCHES_TO_APPLY[@]}" | grep -c "^$TRIMMED$")
+
+# If it matches, prepend the part that interests us to the given commit message
+if [ -n "$BRANCH_NAME" ] && [[ $APPLY_TO_BRANCH -eq 1 ]]; then
+	sed -i.bak -e "1s/^/$TRIMMED: /" $1
+fi
```
Comment (portellaa) [inline hooks/prepare-commit-msg:20]: missing new line

File: hooks/prepare-commit-msg [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+BRANCH_NAME=$(git symbolic-ref --short HEAD)
+
+# Remove the unnecessary parts
+TRIMMED=$(echo $BRANCH_NAME | grep -Eo '^(\w+)' | sed -e 'y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/')
+
+# Check if it matches
+APPLY_TO_BRANCH=$(printf "%s\n" "${BRANCHES_TO_APPLY[@]}" | grep -c "^$TRIMMED$")
+
+# If it matches, prepend the part that interests us to the given commit message
+if [ -n "$BRANCH_NAME" ] && [[ $APPLY_TO_BRANCH -eq 1 ]]; then
```
Comment (portellaa) [inline hooks/prepare-commit-msg:18]: to check if array contains something, you can just do `if [[ ${BRANCHES_TO_APPLY[*]} == *"$BRANCH"* ]]; then`, which replaces the `APPLY_TO_BRANCH` and the `if [ -n "$BRANCH_NAME" ] && [[ $APPLY_TO_BRANCH -eq 1 ]]; then`, but they are both ok

**Review by portellaa — CHANGES_REQUESTED** [review]

**Comment by portellaa** [comment]
i think it misses the validation if the commit message already contains this and also you can't use contexts in the message

File: hooks/prepare-commit-msg (hunk 2 above)
Comment (portellaa) [reply hooks/prepare-commit-msg:12]: you could use `tr`, `awk` or even `sed` with the `L` modifier to lowercase the values

Comment (rafaelcpalmeida) [reply hooks/prepare-commit-msg:12]: Thanks for those! 🔥

**Comment by rafaelcpalmeida** [comment]
Managed to implement the validation whether the user's commit message already includes or not the prefix that the hook would append.

```bash
▶ git commit -m "teste"
[feat/teste b051396] feat: teste
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 h

▶ git commit -m "feat: teste"
[feat/teste 5512f68] feat: teste
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 i

▶ git commit -m "feat(teste): teste"
[feat/teste cf7516f] feat(teste): teste
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 j
```

**Pull request closed without merging**